src/main/java/com/flowdock/plugins/confluence/FlowdockEventRenderer.java

```diff
@@ -25,71 +45,126 @@
+		eventMap.put(CommentCreateEvent.class, "comment_created");
+		eventMap.put(CommentUpdateEvent.class, "comment_updated");
+		eventMap.put(PageCreateEvent.class, "page_created");
+		eventMap.put(PageRemoveEvent.class, "page_removed");
+		eventMap.put(PageRestoreEvent.class, "page_restored");
+		eventMap.put(PageTrashedEvent.class, "page_trashed");
+		eventMap.put(PageUpdateEvent.class, "page_updated");
+	}
+
+	private JsonObject renderEventJson(ContentEvent event, String eventName) {
+		JsonObject result = new JsonObject();
+		result.setProperty("object", renderContentJson(event.getContent()));
+		result.setProperty("event", eventMap.get(event.getClass()));
+		result.setProperty("base_url", getBaseUrl());
 		User user = this.findEventUser(event);
-		result.put("user_email", user.getEmail());
-		result.put("user_name", user.getFullName());
-		
-		if (event instanceof PageCreateEvent) {
-			result.put("event", "create");
-			
-			String content = event.getPage().getBodyAsString();
-			result.put("content_summary", content);
-		} else if (event instanceof PageTrashedEvent) {
-			result.put("event", "delete");
-			
-			String content = event.getPage().getBodyAsString();
-			result.put("content_summary", content);
-		} else if (event instanceof PageUpdateEvent) {
-			result.put("event", "update");
-			result.put("diff", getDiff((PageUpdateEvent)event));
-
-			String content = event.getPage().getBodyAsString();
-			result.put("content_summary", content);	
-		} else {
-			throw new RuntimeException("Unknown page event type.");
+		if (user != null)
+			result.setProperty("user", renderUserData(user));
+		return result;
+	}
+
+	private JsonObject renderContentJson(ContentEntityObject content) {
+		if (content instanceof Page)
+			return renderPageData((Page) content);
+		else if (content instanceof BlogPost)
+			return renderBlogData((BlogPost) content);
+		else if (content instanceof Comment)
+			return renderCommentData((Comment) content);
+		else
+			throw new RuntimeException("Unknown event content");
+	}
+
+	private JsonObject renderSpaceData(Space space) {
+		JsonObject result = new JsonObject();
+		result.setProperty("name", space.getName());
+		result.setProperty("url", getBaseUrl() + space.getUrlPath());
+		return result;
+	}
+
+	private JsonObject renderBlogData(BlogPost blog) {
+		JsonObject result = renderObjectData(blog);
+		result.setProperty("space", renderSpaceData(blog.getSpace()));
+		return result;
+	}
+
+	private JsonObject renderCommentData(Comment comment) {
+		JsonObject result = renderObjectData(comment);
+		if (comment.getParent() != null)
+			result.setProperty("parent", renderObjectData(comment.getParent()));
+		result.setProperty("container", renderContentJson(comment.getContainer()));
+		return result;
+	}
+
+	private JsonObject renderPageData(Page page) {
+		JsonObject result = renderObjectData(page);
+		JsonArray ancestors = new JsonArray();
+		for (Iterator<Page> i = page.getAncestors().iterator(); i.hasNext();) {
```

Comment:
You should be able to use an [enhanced for loop](http://www.javacertifications.net/javacert/enhancedforloop.jsp) here. Something like..

```java
for(Page ancestor : page.getAncestors) {
}
```